hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/response/snapshot/OMSnapshotPurgeResponse.java

```diff
@@ -110,9 +119,18 @@ private void cleanupSnapshotChain(OmMetadataManagerImpl metadataManager,
 
       String snapshotTableKey = snapshotChainManager
           .getTableKey(nextGlobalSnapshotId);
-      SnapshotInfo nextGlobalSnapInfo =
-          metadataManager.getSnapshotInfoTable().get(snapshotTableKey);
-      if (nextGlobalSnapInfo != null) {
+      nextGlobalSnapInfo = metadataManager.getSnapshotInfoTable()
+          .get(snapshotTableKey);
+      // If both next global and path snapshot are same, it may overwrite
+      // nextPathSnapInfo.setPathPreviousSnapshotID(), adding this check
+      // will prevent it.
+      if (nextGlobalSnapInfo != null && nextGlobalSnapInfo.getSnapshotID()
+          .equals(nextPathSnapInfo.getSnapshotID())) {
+        nextPathSnapInfo.setGlobalPreviousSnapshotID(
+            snapInfo.getPathPreviousSnapshotID());
+        metadataManager.getSnapshotInfoTable().putWithBatch(batchOperation,
+            nextPathSnapInfo.getTableKey(), nextPathSnapInfo);
+      } else if (nextGlobalSnapInfo != null) {
```

Comment:
In this case described in the comment, we need to update both `nextPathSnapInfo` and `nextGlobalSnapInfo` right? if so we shouldn't use `else if` imo

Reply:
The reason for using `else if` because `nextGlobalSnapInfo` can be `null` if we are deleting the last snapshot, so we need that check.